pkg/plugins/client_test.go

```diff
@@ -38,6 +39,26 @@ func TestFailedConnection(t *testing.T) {
 	}
 }
 
+func TestFailOnce(t *testing.T) {
+	addr := setupRemotePluginServer()
+	defer teardownRemotePluginServer()
+
+	failed := false
+	mux.HandleFunc("/Test.FailOnce", func(w http.ResponseWriter, r *http.Request) {
+		if !failed {
+			failed = true
+			panic("Plugin not ready")
```

Comment:
I tried, but it will cause the test to always fail, even if the retry logic succeeds.
If you have a less brutal option than `panic`, I'm all for it.